### PR TITLE
Java 24276 Upgrade Keycloak for oauth-rest/keycloak-custom-providers

### PR DESCRIPTION
**Pull request by anuragkumawat**

### CODE REVIEW

File: oauth-rest/keycloak-custom-providers/src/test/java/com/baeldung/auth/provider/user/ContextIntegrationLiveTest.java [hunk 1]
```diff
@@ -20,14 +25,8 @@
...
 
-import static org.junit.jupiter.api.Assertions.*;
-
-import javax.sql.DataSource;
-
-import org.junit.jupiter.api.BeforeAll;
-
 @ExtendWith(SpringExtension.class)
-@SpringBootTest(classes = { AuthorizationServerApp.class }, webEnvironment = WebEnvironment.RANDOM_PORT)
+@SpringBootTest(classes = { AuthorizationServerApp.class }, webEnvironment = WebEnvironment.DEFINED_PORT)
```
Comment (dkapil) [inline oauth-rest/keycloak-custom-providers/src/test/java/com/baeldung/auth/provider/user/ContextIntegrationLiveTest.java:29]: can this be converted to integration test?

File: oauth-rest/keycloak-custom-providers/src/main/java/com/baeldung/auth/provider/user/CustomUser.java [hunk 2]
```diff
@@ -121,4 +122,9 @@ CustomUser build() {
             
         }
     }
+
+    @Override
+    public SubjectCredentialManager credentialManager() {
+        return new LegacyUserCredentialManager(session, realm, this);
```
Comment (dkapil) [inline oauth-rest/keycloak-custom-providers/src/main/java/com/baeldung/auth/provider/user/CustomUser.java:128]: can we use other implementation of SubjectCredentialManager

**Review by dkapil — CHANGES_REQUESTED** [review]

File: oauth-rest/keycloak-custom-providers/src/main/java/com/baeldung/auth/provider/user/CustomUser.java (hunk 2 above)
Comment (anuragkumawat) [reply oauth-rest/keycloak-custom-providers/src/main/java/com/baeldung/auth/provider/user/CustomUser.java:128]: SubjectCredentialManager provides two implementations LegacyUserCredentialManager and MapUserCredentialManager. 
LegacyUserCredentialManager uses UserStorageProvider and MapUserCredentialManager uses MapUserEntity.
Our code is built around UserStorageProvider implementation, hence I have used LegacyUserCredentialManager.

File: oauth-rest/keycloak-custom-providers/src/test/java/com/baeldung/auth/provider/user/ContextIntegrationLiveTest.java (hunk 1 above)
Comment (anuragkumawat) [reply oauth-rest/keycloak-custom-providers/src/test/java/com/baeldung/auth/provider/user/ContextIntegrationLiveTest.java:29]: Changes done. Thanks.

**Pull request merged**